src/styles/buildTextStyles.js

```diff
@@ -99,4 +99,80 @@ const buildTextStyles = css`
   ${({ uppercase }) => (uppercase ? 'text-transform: uppercase;' : '')}
 `;
 
+buildTextStyles.object = ({
+  color,
+  theme,
+  size,
+  isEmoji,
+  family,
+  mono,
+  weight,
+  letterSpacing,
+  lineHeight,
+  opacity,
+  align,
+  tabularNums,
+  uppercase,
+}) => {
+  const styles = {
+    color: colors.get(color, theme.colors) || theme.colors.dark,
+  };
+
+  // function is used because of default argument values
+  (({ isEmoji, family = 'SFProRounded', mono, weight }) => {
+    if (isEmoji) {
+      return;
+    }
+
+    styles.fontFamily = familyFontWithAndroidWidth(weight, family, mono);
+  })({ family, isEmoji, mono, weight });
+
+  // function is used because of default argument values
+  (({ size = 'medium' }) => {
+    styles.fontSize =
+      typeof size === 'number' ? size : fonts.size[size] ?? size;
+  })({
+    size,
+  });
+
+  // function is used because of default argument values
+  (({ isEmoji, weight = 'regular' }) => {
+    if (isEmoji || isNil(weight) || android) {
+      return;
+    }
+
+    styles.fontWeight = fonts.weight[weight] ?? weight;
+  })({ isEmoji, weight });
+
+  // function is used because of default argument values
+  (({ letterSpacing = 'rounded' }) => {
+    if (!isNil(letterSpacing)) {
+      styles.letterSpacing =
+        fonts.letterSpacing[letterSpacing] ?? letterSpacing;
+    }
+  })({ letterSpacing });
+
+  if (!(isNil(lineHeight) || (isEmoji && android))) {
```

Comment:
I don't think this logic matches the original implementation which (naively inverted) is `!(isNil(lineHeight) || (isEmoji && android))`